drivers/modem/hl7800.c

```diff
@@ -1070,6 +1072,32 @@ void mdm_hl7800_generate_status_events(void)
 	hl7800_unlock();
 }
 
+/** @ref /zephyr/samples/subsys/logging/logger/src/main.c */
+static int log_source_id_get(const char *name)
+{
+	for (int i = 0; i < log_src_cnt_get(CONFIG_LOG_DOMAIN_ID); i++) {
+		if (strcmp(log_source_name_get(CONFIG_LOG_DOMAIN_ID, i),
+			   name) == 0) {
+			return i;
+		}
+	}
+	return -1;
+}
+
+uint32_t mdm_hl7800_log_filter_set(uint32_t level)
```

Comment:
Who's the user of this function, responsible for changeing the log level? I see in the commit message that the `shell` is involved - why not use the shell module provided by the logger to change the log level?

Reply:
I will fix the commit message. The goal is that any app can change the log level output at runtime even if the shell isn't being used.